File: EPRO.Library.v3.5/EPRO.Library.v3.5/IEqualityComparerIgnoreCase.cs
using System.Collections.Generic;
using ELibrary.Standard.Modules;

namespace ELibrary.Standard.Objects
{
    public class IEqualityComparerIgnoreCase : IEqualityComparer<string>
    {

        #region IEquality Comparer - Ignore Case
        bool IEqualityComparer<string>.Equals(string x, string y)
        {
            return x.equalsIgnoreCase(y);
        }

        public bool Equals1(string x, string y) => ((IEqualityComparer<string>)this).Equals(x, y);

        int IEqualityComparer<string>.GetHashCode(string obj)
        {
            return obj.GetHashCode();
        }

        public int GetHashCode1(string obj) => ((IEqualityComparer<string>)this).GetHashCode(obj);

        #endregion

    }
}

File: EPRO.Library.v3.5/EPRO.Library.v3.5/IEqualityComparerIgnoreCase.vb
Imports EPRO.Library.v3._5.Modules

Namespace Objects


    Public Class IEqualityComparerIgnoreCase
        Implements IEqualityComparer(Of String)

#Region "IEquality Comparer - Ignore Case"
        Public Function Equals1(x As String, y As String) As Boolean Implements IEqualityComparer(Of String).Equals
            Return equalsIgnoreCase(x, y)
        End Function

        Public Function GetHashCode1(obj As String) As Integer Implements IEqualityComparer(Of String).GetHashCode
            Return obj.GetHashCode()
        End Function

#End Region

    End Class

End Namespace
